repart: Allow specifying --copy-from more than once

DaanDeMeyer opened the pull request:
Definitions will be synthesized from each of the given images.

bluca commented:
Can you add some test coverage for this please?

DaanDeMeyer commented:
@bluca We already have full coverage with the existing test. This just executes the exact same logic if you specify more than two sources.

bluca commented:
I know, but having a sanity check to ensure it works correctly when reading from 2 images would be nice, no? For ASAN coverage and peace of mind

poettering commented:
lgtm. thanks for addressing this so quickly

DaanDeMeyer commented:
Updated integration test and added a trivial extra commit to always acquire a shared lock on the nodes passed to --copy-from as we never write to these

bluca commented on test/units/testsuite-58.sh:
```diff
@@ -160,12 +160,24 @@ last-lba: 2097118
                    --seed="$seed" \
-                   --copy-from="$imgs/zzz" \
+                   --definitions "" \
+                   --copy-from="$imgs/qqq" \
```
zzz?

bluca replied:
but then it's the same one used twice?

DaanDeMeyer replied:
Yeah, we just copy.the partitions from that one twice, that exercises the logic two times.
